src/Framework/BuildSubmissionStartedEventArgs.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Framework
+{
+    [Serializable]
+    public class BuildSubmissionStartedEventArgs : BuildStatusEventArgs
+    {
+        public IReadOnlyDictionary<string, string?>? GlobalProperties { get; protected set; }
+
+        public IEnumerable<string>? EntryProjectsFullPath { get; protected set; }
+
+        public ICollection<string>? TargetNames { get; protected set; }
+
+        public BuildRequestDataFlags? Flags { get; protected set; }
+
+        public int? SubmissionId { get; protected set; }
```

Comment:
I added the nullable because of this function: https://github.com/dotnet/msbuild/blob/8b03742aafd18a26a5d03cd045e4d24963886503/src/Shared/LogMessagePacketBase.cs#L608 Where all the events are initialized with default or null constructors. I'm honestly not really sure in what part of the logging this falls on, but if there is a way around it I can make the properties non nullable.